cmd/testbucket/plan.go

```
@@ -0,0 +1,651 @@
+// report built from dozens of Fprintf calls can be checked once at the end
+// instead of threading an error through every line.
+type errWriter struct {
+	w   io.Writer
+	err error
+}
+
+func (e *errWriter) Write(p []byte) (int, error) {
+	if e.err != nil {
+		// Report the length as written: the caller is mid-report and the
+		// first error is the one worth keeping.
+		return len(p), nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 490) but it returns nil

(nilerr)